torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -229,231 +231,228 @@ class TensorExprFuser {
+      }
+      subgraph->appendNode(new_n);
     }
-    GRAPH_UPDATE("Creating a tensorexpr::Group node from: ", *n);
-    return aliasingSafeSubgraphMerge(
-        n, [&](std::unordered_map<Value*, Value*>& vmap) {
-          return SubgraphUtils::createSingletonSubgraph(
-              n, prim::TensorExprGroup, vmap);
-        });
-  }
 
-  void mergeNodeIntoSubgraphAndUpdateAliasing(Node* n, Node* subgraph) {
-    aliasingSafeSubgraphMerge(n, [&](std::unordered_map<Value*, Value*>& vmap) {
-      SubgraphUtils::mergeNodeIntoSubgraph(n, subgraph, vmap);
-      return subgraph;
-    });
-  }
+    std::map<Value*, size_t> index_map;
+    for (const auto& v : outputs) {
+      index_map[v] = subgraph->registerOutput(value_map.at(v));
+      auto output = subgraph_node->addOutput();
+      output->setType(v->type());
+    }
 
-  // Add unvisited input nodes to the queue for further merging into the fusion
-  // group.
-  void updateQueue(
-      Node* fusion_group,
-      std::set<Node*, nodesComparator>& queue,
-      const std::unordered_set<Node*>& visited) {
-    for (auto input : fusion_group->inputs()) {
-      if (!visited.count(input->node())) {
-        queue.insert(input->node());
+    Node* insert_point = inputs.front()->node();
+    for (const auto& input : inputs) {
+      if (input->node()->isAfter(insert_point)) {
+        insert_point = input->node();
       }
     }
-  }
-
-  // Create a fusion group starting from the node N.
-  // We then try to pull inputs into the fusion group and repeat that process
-  // until there is nothing we can pull in.
-  Node* createFusionGroup(Node* n) {
-    // Queue of the nodes we should consider for merging into the fusion groups
-    // (those nodes are usually inputs of the fusion group).
-    // We use an ordered set here to visit them in the right order: the fusion
-    // group is closer to the end of the block and we are trying to pull later
-    // nodes first.
-    // NB: the order in the list in theory could stale if we move nodes around.
-    // However, this should only happen to the nodes we could not fuse, and
-    // hence it should not be a problem.
-    std::set<Node*, nodesComparator> queue;
-    std::unordered_set<Node*> visited_nodes;
 
-    Node* fusion_group = n;
-    if (min_group_size_ == 1) {
-      fusion_group = getOrCreateTensorExprSubgraph(n);
+    Node* after_point = outputs.front()->node();
+    for (const auto& output : outputs) {
+      if (output->node()->isBefore(after_point)) {
+        after_point = output->node();
+      }
     }
 
-    updateQueue(fusion_group, queue, visited_nodes);
-
-    GRAPH_DEBUG("Iteratively pull input nodes into the fusion group...\n");
-    while (!queue.empty()) {
-      debugDumpFusionGroup("Current fusion group: ", fusion_group);
-      GRAPH_DEBUG(queue.size(), " nodes are in the queue.\n");
-
-      Node* input_node = *queue.begin();
-      queue.erase(queue.begin());
+    // TODO This is not true in the general case!!
```

Comment:
[correctness] I need to insert and the topo-sort to be correct.  The search algorithm can find cases where the last input is after the first user (not topologically, but with respect to the current ordering)